### ISSUE TITLE
Crash on startup when using the GLES3 backend

### ISSUE THREAD

**Issue by Geometror** [godotengine/godot#63600]
### Godot version

4.0dev

### System information

Windows 10, GLES3 backend

### Issue description

The engine crashes directly when loading the editor/a project with 3d content (using the GLES3 backend).
Bisected and it seems like it was caused by #63345.

```
Crashlog:
================================================================
CrashHandlerException: Program crashed
Engine version: Godot Engine v4.0.alpha.custom_build (14d021287bced6a7f5ab9db24936bd07b4cfdfd0)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[0] SelfList<RasterizerSceneGLES3::GeometryInstanceGLES3>::List::add (C:\Users\hendr\Desktop\Godot\dev\main\godot\core\templates\self_list.h:49)
[1] RasterizerSceneGLES3::GeometryInstanceGLES3::_mark_dirty (C:\Users\hendr\Desktop\Godot\dev\main\godot\drivers\gles3\rasterizer_scene_gles3.cpp:124)
[2] RasterizerSceneGLES3::geometry_instance_create (C:\Users\hendr\Desktop\Godot\dev\main\godot\drivers\gles3\rasterizer_scene_gles3.cpp:58)
[3] RendererSceneCull::instance_set_base (C:\Users\hendr\Desktop\Godot\dev\main\godot\servers\rendering\renderer_scene_cull.cpp:638)
[4] RenderingServerDefault::instance_set_base (C:\Users\hendr\Desktop\Godot\dev\main\godot\servers\rendering\rendering_server_default.h:745)
[5] VisualInstance3D::set_base (C:\Users\hendr\Desktop\Godot\dev\main\godot\scene\3d\visual_instance_3d.cpp:132)
[6] SpriteBase3D::SpriteBase3D (C:\Users\hendr\Desktop\Godot\dev\main\godot\scene\3d\sprite_3d.cpp:437)
[7] AnimatedSprite3D::AnimatedSprite3D (C:\Users\hendr\Desktop\Godot\dev\main\godot\scene\3d\sprite_3d.cpp:1292)
[8] ClassDB::creator<AnimatedSprite3D> (C:\Users\hendr\Desktop\Godot\dev\main\godot\core\object\class_db.h:139)
[9] ClassDB::instantiate (C:\Users\hendr\Desktop\Godot\dev\main\godot\core\object\class_db.cpp:340)
[10] ClassDB::class_get_default_property_value (C:\Users\hendr\Desktop\Godot\dev\main\godot\core\object\class_db.cpp:1436)
[11] get_documentation_default_value (C:\Users\hendr\Desktop\Godot\dev\main\godot\editor\doc_tools.cpp:320)
[12] DocTools::generate (C:\Users\hendr\Desktop\Godot\dev\main\godot\editor\doc_tools.cpp:417)
[13] EditorHelp::generate_doc (C:\Users\hendr\Desktop\Godot\dev\main\godot\editor\editor_help.cpp:1961)
[14] EditorNode::EditorNode (C:\Users\hendr\Desktop\Godot\dev\main\godot\editor\editor_node.cpp:5944)
[15] Main::start (C:\Users\hendr\Desktop\Godot\dev\main\godot\main\main.cpp:2508)
[16] widechar_main (C:\Users\hendr\Desktop\Godot\dev\main\godot\platform\windows\godot_windows.cpp:176)
[17] _main (C:\Users\hendr\Desktop\Godot\dev\main\godot\platform\windows\godot_windows.cpp:201)
[18] main (C:\Users\hendr\Desktop\Godot\dev\main\godot\platform\windows\godot_windows.cpp:215)
[19] WinMain (C:\Users\hendr\Desktop\Godot\dev\main\godot\platform\windows\godot_windows.cpp:229)
[20] __scrt_common_main_seh (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[21] BaseThreadInitThunk
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

Open any project with the driver set to OpenGL (3d has to be enabled since the bug is related to GeometryInstance3D).

### Minimal reproduction project

-

**Comment by BastiaanOlij** [comment]
I'll have a look at this in the morning, probably accessing something in the wrong order. A lot has changed in this code and I've only tested fully on Render Device. @clayjohn if you have a chance to look that be great.

**Comment by BastiaanOlij** [comment]
Should be fixed by #63761
